### ISSUE TITLE
Ideas for future ez addons

### ISSUE THREAD

**Issue by asirota** [NewPath-Consulting/watm#117]
There are [many great ideas](https://forums.wildapricot.com/forums/308932-wishlist/filters/top?fbclid=IwAR0LCKWt3W8_SAJFLSvbnrEHgHb1vxUQbQhpae4LaiOYUgOMo1EwMIJ-OYg) requested and voted by wildapricot customers. 

Some particularly interesting ones that can be scoped down and implemented with an ez macro

https://forums.wildapricot.com/forums/308932-wishlist/suggestions/33345121-online-forms

https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825707-fast-registration-or-simplify-buying-registrations

https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825554-members-to-be-able-to-submit-events

https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825830-installments-payment-plans-for-events-and-member

https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825629-discount-coupons-for-event-registrations

https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825950-customize-event-registration-form-for-each-reg-ty

https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8826118-registering-families-bundles-for-events

https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825575-limit-number-of-selection-for-individual-registrat

**Comment by asirota** [comment]
Another useful one layout document library better

https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825557-document-library-gadget

**Comment by asirota** [comment]
This one could be quite easy

https://forums.wildapricot.com/forums/308932-wishlist/suggestions/10091643-maps-for-venue-location

**Comment by mDisna** [comment]
> https://forums.wildapricot.com/forums/308932-wishlist/suggestions/33345121-online-forms
> 
> https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825707-fast-registration-or-simplify-buying-registrations
> 
> https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825554-members-to-be-able-to-submit-events
> 
> https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825830-installments-payment-plans-for-events-and-member
> 
> https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825629-discount-coupons-for-event-registrations
> 
> https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825950-customize-event-registration-form-for-each-reg-ty
> 
> https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8826118-registering-families-bundles-for-events
> 
> https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825575-limit-number-of-selection-for-individual-registrat

These can't be implemented at all or can't be implemented with JS alone (would require server-side code)

**Comment by mDisna** [comment]
> This one could be quite easy
> 
> https://forums.wildapricot.com/forums/308932-wishlist/suggestions/10091643-maps-for-venue-location

I've made a customization before that will automatically create a map of the address from an event so that's possible - however, it won't appear in the email.

I was also thinking of making a macro for inserting static maps into content gadgets.

Both of these options however will require a Google Maps API key

**Comment by asirota** [comment]
Creating masks on certain type of fields is made for ez

https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8826766-database-fields-custom-validation-e-g-regular-e

**Comment by mDisna** [comment]
> Strange one but potentially easy to do
> 
> https://forums.wildapricot.com/forums/308932-wishlist/suggestions/11710026-link-posts-and-events-to-social-media-channels

If I'm understanding what they are asking for, this is not possible.

**Comment by asirota** [comment]
> These can't be implemented at all or can't be implemented with JS alone (would require server-side code)

We have admin API access through Make.com. Scenarios could be deployed along side the EZ addons and work in tandem using webhooks. We have already implemented Make webhooks to check the EZ license. 

Example for the "custom forms" feature: EZ could render a simple form -> calls a Make.com scenario which populates the WA database with data.

So all of them are possible with some Make.com support. Make.com scenarios can be deployed using the [Make.com API](https://www.make.com/en/api-documentation) and [app ](https://www.make.com/en/help/app/make) as well. It's time to take a look at that closer I think.

**Comment by asirota** [comment]
> If I'm understanding what they are asking for, this is not possible.

yeah I misread this one, this has to do with background syncing that can be done with Make easily enough on a schedule but not a solution for EZ which involves mostly members and anonymous visitors to the site.

**Comment by asirota** [comment]
> If I'm understanding what they are asking for, this is not possible.

I depends on how you interpret it. Converting something that looks like an address to a link to google maps/apple maps is easy enough -- it's like a scanner. Converting it to an embeddable map is harder and requires a key and I would not implement that. People just want a hyper link that appears in "locations" automatically.

**Comment by asirota** [comment]
> Another useful one layout document library better
> 
> https://forums.wildapricot.com/forums/308932-wishlist/suggestions/8825557-document-library-gadget

This one was implemented as a customizable library gadget by linking to existing files in the File Manager in 2.0.5

**Comment by asirota** [comment]
> I was also thinking of making a macro for inserting static maps into content gadgets.

Don't bother here -- SimpleMaps.com already does this well. The dynamic non-Google maps is a lot more complex than it seems at first.

**Comment by asirota** [comment]
In general add ons can take lots of client side rendering features that are just extensions of the standard Content Gadgets:

- decorating content that is easier to navigate/read
- linking address links automatically
- linking email addresses and URLs automatically

The forms idea can be quite generic.

[ez-forms submitbutton="text name" targetwebhook="MyMake.com Webhook"]
[field1 type=text|password|checkbox|radio|select|longtext|date mandatory=true|false label="field1 label" waintegration=wafieldname]
[field2 type=text|password|checkbox|radio|select|longtext|date mandatory=true|false label="field2 label" waintegration=fieldname]
[/ez-forms]

On the make side we would publish a scenario that could accept this data including the wafieldname bindings and the end user can program it to route into WildApricot. The scenario could be generic enough to make it easy to change. The benefits are many (native form support, integration with WA data). It's a hugely important feature.
